draconic/helpers.py

```diff
@@ -279,6 +279,16 @@ def union(self, *s):
                 _raise_in_context(IterableTooLong, "This set is too large")
             return SafeSet(super().union(*s))
 
+        def intersection(self, *s):
+            if approx_len_of(self) + sum(approx_len_of(other) for other in s) > config.max_const_len:
+                _raise_in_context(IterableTooLong, "This set is too large")
+            return SafeSet(super().union(*s))
```

Comment:
this should probably be `super.intersection`

Reply:
also, intersection's size check shouldn't be the sum, as the size of the intersection of two sets is guaranteed to be equal to or less than the size of the smaller set. It should check each set individually to make sure each set individually is not too large.